src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -1195,33 +733,20 @@ where
                     // past the input AFTER the output capability of this operator itself has
                     // been downgraded (which drives the data shard upper), AND the
                     // remap shard has been advanced below.
-                    if let Some(new_compaction_since) = derive_new_compaction_since(
-                        resumption_frontier,
-                        &last_compaction_since,
+                    let upper_ts = resume_upper.as_option().copied();
+                    if let Some(upper_ts) = upper_ts {
                         // Choose a `since` as aggresively as possible
-                        1,
-                        id,
-                        "remap",
-                        worker_id,
-                        worker_count,
-                    ) {
-                        timestamper.compact(new_compaction_since.clone()).await;
-                        last_compaction_since = new_compaction_since;
-                    }
-                }
-                Some(event) = input.next() => match event {
-                    Event::Data(_cap, data) => {
-                        data.swap(&mut buffer);
-                        for batch_upper_summary in buffer.drain(..) {
-                            for (pid, offset) in batch_upper_summary.batch_upper.iter() {
-                                global_source_upper.maybe_insert(pid.clone(), *offset);
-                            }
+                        let compaction_since = Antichain::from_elem(upper_ts.saturating_sub(1));
```

Comment:
are we sure we don't want to keep the function? having a trace log line is pretty nice...